cpp/src/arrow/compute/api_scalar.h

```diff
@@ -269,6 +270,31 @@ struct ARROW_EXPORT DayOfWeekOptions : public FunctionOptions {
   uint32_t week_start;
 };
 
+/// Used to control timestamp timezone conversion and handling ambiguous/nonexistent
+/// times.
+struct ARROW_EXPORT TemporalLocalizationOptions : public FunctionOptions {
+ public:
+  /// How to interpret ambiguous local times that can be interpreted as
+  /// multiple instants due to DST shifts.
+  enum Ambiguous { AMBIGUOUS_RAISE, AMBIGUOUS_EARLIEST, AMBIGUOUS_LATEST };
+
+  /// How to handle local times that do not exists due to DST shifts.
+  enum Nonexistent { NONEXISTENT_RAISE, NONEXISTENT_EARLIEST, NONEXISTENT_LATEST };
```

Comment:
@bkietz @jorisvandenbossche  Any thoughts about naming? Should these be called `AmbiguousHandling` and `NonExistentHandling`?